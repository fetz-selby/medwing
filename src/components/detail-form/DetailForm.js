import React, {Component} from 'react';
import AutoSuggest from 'react-autosuggest';
import PropTypes from 'prop-types';
import './detail-form.css';

class DetailForm extends Component {
        state = {
            suggested: ''
        }

    // When the compoennt is loaded for the first time.
    componentWillMount() {
        console.log('componentWillMount => '+JSON.stringify(this.props.location));
        if (this.props.location) this.setDefaultState(this.props.location);
    }

    // When the component is loaded again.
    componentWillReceiveProps(nextProps) {
        this.setDefaultState(nextProps.location);
    }

    // Prefill input fields with the available data by setting default state.
    setDefaultState(location) {

        // Set state using data.
        this.setState({
            id: location.id,
            title: location.title,
            address: location.address,
            lat: location.lat,
            lng: location.lng,
            suggested: location.address?location.address:''
        })
    }

    titleChangeHandler=(evt, newValue)=>{
        this.setState({
            title: newValue
        })
    }

    onSuggestedChange = (evt, { newValue }) =>{
        this.setState({
            suggested: newValue
        });
    }

    onSuggestionSelected = () =>{
        this.setState({
            lat: this.selectedSuggestion.lat,
            lng: this.selectedSuggestion.lng,
            address: this.selectedSuggestion.address
        })
    }

    onSuggestionsClearRequested = () => {
        this.props.clearSuggestions();
    };

    onSuggestionsFetchRequested = ({value}) => {
        this.props.fetchSuggestions(value);
    }

    getSuggestionValue = suggestion => {this.selectedSuggestion = {...suggestion}; return suggestion.address};

    renderSuggestion = suggestion => (
        <div>
        {suggestion.address}
        </div>
    );

    render(){
        const {suggested} = this.state;
        const {suggestions, isUpdate} = this.props;
    
        const inputProps = {
            placeholder: 'Enter Address',
            value: suggested,
            onChange: this.onSuggestedChange
        };


        return <div className='detail-form'>

            <div className='row'>
                <div className='label'>Title</div>
                <div className='clearfix'></div>
                <input type='text' className='inputs' value={this.state.title} onChange={this.titleChangeHandler} />
            </div>
            <div className='clearfix'></div>

            <div className='row'>
                <div className='label'>Address</div>
                <div className='clearfix'></div>
                <AutoSuggest 
                    suggestions={suggestions}
                    onSuggestionsFetchRequested={this.onSuggestionsFetchRequested}
                    onSuggestionsClearRequested={this.onSuggestionsClearRequested}
                    getSuggestionValue={this.getSuggestionValue}
                    renderSuggestion={this.renderSuggestion}
                    highlightFirstSuggestion={true}
                    inputProps={inputProps}
                    onSuggestionSelected={this.onSuggestionSelected}
                />
            </div>
            <div className='clearfix'></div>

            <div className='row'>
                <div className='label'>Latitude</div>
                <div className='clearfix'></div>
                <input type='text' className='inputs' disabled={true} value={this.state.lat}/>
            </div>
            <div className='clearfix'></div>

            <div className='row'>
                <div className='label'>Longitude</div>
                <div className='clearfix'></div>
                <input type='text' className='inputs' disabled={true} value={this.state.lng}/>
            </div>

            <div className='row'>
                <div className={isUpdate?'delete-btn':'delete-btn light-hide'}>Delete</div>
                <button className='save-btn'>{isUpdate ? 'Update':'Save'}</button>
            </div>
        </div>
    }
}

DetailForm.propTypes = {
    suggestion: PropTypes.array,
    location: PropTypes.object,
    isUpdate: PropTypes.bool.isRequired,
    clearSuggestions: PropTypes.func,
    fetchSuggestions: PropTypes.func,
}

export default DetailForm;