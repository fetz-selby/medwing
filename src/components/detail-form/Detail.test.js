import React from 'react';
import { shallow } from 'enzyme';
import DetailForm from './DetailForm';

describe('Detail Form Component', ()=>{
    it('should render [SAVE] on button', ()=>{
        const location = {id: 0, title:'', address: '', lat: '', lng: ''};
        const currentLocationState = jest.fn;
        const onDelete = jest.fn;

        const props ={
            location,
            isUpdate: false,
            currentLocationState,
            onDelete
        }
        
        const df = shallow(<DetailForm {...props}/>);
        expect(df.find('.save-btn').text()).toBe('Save');
    })

    it('should render [UPDATE] on button', ()=>{
        const location = {id: 1, title:'test 1', address: 'street 12', lat: '51.165691', lng: '10.451526000000058'};
        const currentLocationState = jest.fn;
        const onDelete = jest.fn;

        const props ={
            location,
            isUpdate: true,
            currentLocationState,
            onDelete
        }
        
        const df = shallow(<DetailForm {...props}/>);
        expect(df.find('.save-btn').text()).toBe('Update');
    })

    it('should expect props.onUpdate on update clicked', ()=>{
        const location = {id: 1, title:'test 1', address: 'street 12', lat: '51.165691', lng: '10.451526000000058'};
        const currentLocationState = jest.fn;
        const onDelete = jest.fn;
        const onUpdate = jest.fn;

        const props ={
            isUpdate: true,
            currentLocationState,
            onUpdate,
            onDelete,
            location
        }

        const df = shallow(<DetailForm {...props}/>);
        df.find('.save-btn').simulate('click');
        expect(onUpdate).toHaveBeenCalled();
    })

    it('should render correctly', ()=>{
        const location = {id: 1, title:'test 1', address: 'street 12', lat: '51.165691', lng: '10.451526000000058'};
        const currentLocationState = jest.fn;
        const onDelete = jest.fn;
        const onUpdate = jest.fn;

        const props ={
            isUpdate: true,
            currentLocationState,
            onUpdate,
            onDelete,
            location
        }

        const df = shallow(<DetailForm {...props}/>);
        expect(df).toMatchSnapshot();
    })
})