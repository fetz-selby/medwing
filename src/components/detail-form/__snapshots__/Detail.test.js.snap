// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Detail Form Component should render correctly 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DetailForm
    currentLocationState={[Function]}
    isUpdate={true}
    location={
      Object {
        "address": "street 12",
        "id": 1,
        "lat": "51.165691",
        "lng": "10.451526000000058",
        "title": "test 1",
      }
    }
    onDelete={[MockFunction]}
    onUpdate={[MockFunction]}
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <div
          className="row"
        >
          <div
            className="display light-hide"
          >
            No Message
          </div>
        </div>,
        <div
          className="clearfix"
        />,
        <div
          className="row"
        >
          <div
            className="label"
          >
            Title
          </div>
          <div
            className="clearfix"
          />
          <input
            className="inputs"
            onChange={[Function]}
            type="text"
            value="test 1"
          />
        </div>,
        <div
          className="clearfix"
        />,
        <div
          className="row"
        >
          <div
            className="label"
          >
            Address
          </div>
          <div
            className="clearfix"
          />
          <Autosuggest
            alwaysRenderSuggestions={false}
            focusInputOnSuggestionClick={true}
            getSuggestionValue={[Function]}
            highlightFirstSuggestion={true}
            id="1"
            inputProps={
              Object {
                "onChange": [Function],
                "placeholder": "Enter Address",
                "value": "street 12",
              }
            }
            multiSection={false}
            onSuggestionSelected={[Function]}
            onSuggestionsClearRequested={[Function]}
            onSuggestionsFetchRequested={[Function]}
            renderSuggestion={[Function]}
            renderSuggestionsContainer={[Function]}
            shouldRenderSuggestions={[Function]}
            theme={
              Object {
                "container": "react-autosuggest__container",
                "containerOpen": "react-autosuggest__container--open",
                "input": "react-autosuggest__input",
                "inputFocused": "react-autosuggest__input--focused",
                "inputOpen": "react-autosuggest__input--open",
                "sectionContainer": "react-autosuggest__section-container",
                "sectionContainerFirst": "react-autosuggest__section-container--first",
                "sectionTitle": "react-autosuggest__section-title",
                "suggestion": "react-autosuggest__suggestion",
                "suggestionFirst": "react-autosuggest__suggestion--first",
                "suggestionHighlighted": "react-autosuggest__suggestion--highlighted",
                "suggestionsContainer": "react-autosuggest__suggestions-container",
                "suggestionsContainerOpen": "react-autosuggest__suggestions-container--open",
                "suggestionsList": "react-autosuggest__suggestions-list",
              }
            }
          />
        </div>,
        <div
          className="clearfix"
        />,
        <div
          className="row"
        >
          <div
            className="label"
          >
            Latitude
          </div>
          <div
            className="clearfix"
          />
          <input
            className="inputs"
            disabled={true}
            type="text"
            value="51.165691"
          />
        </div>,
        <div
          className="clearfix"
        />,
        <div
          className="row"
        >
          <div
            className="label"
          >
            Longitude
          </div>
          <div
            className="clearfix"
          />
          <input
            className="inputs"
            disabled={true}
            type="text"
            value="10.451526000000058"
          />
        </div>,
        <div
          className="clearfix"
        />,
        <div
          className="row"
        >
          <div
            className="delete-btn"
            onClick={[Function]}
          >
            Delete
          </div>
          <button
            className="save-btn"
            onClick={[Function]}
          >
            Update
          </button>
        </div>,
      ],
      "className": "detail-form",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": <div
            className="display light-hide"
          >
            No Message
          </div>,
          "className": "row",
        },
        "ref": null,
        "rendered": Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "No Message",
            "className": "display light-hide",
          },
          "ref": null,
          "rendered": "No Message",
          "type": "div",
        },
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "clearfix",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="label"
            >
              Title
            </div>,
            <div
              className="clearfix"
            />,
            <input
              className="inputs"
              onChange={[Function]}
              type="text"
              value="test 1"
            />,
          ],
          "className": "row",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Title",
              "className": "label",
            },
            "ref": null,
            "rendered": "Title",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "clearfix",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "inputs",
              "onChange": [Function],
              "type": "text",
              "value": "test 1",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "clearfix",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="label"
            >
              Address
            </div>,
            <div
              className="clearfix"
            />,
            <Autosuggest
              alwaysRenderSuggestions={false}
              focusInputOnSuggestionClick={true}
              getSuggestionValue={[Function]}
              highlightFirstSuggestion={true}
              id="1"
              inputProps={
                Object {
                  "onChange": [Function],
                  "placeholder": "Enter Address",
                  "value": "street 12",
                }
              }
              multiSection={false}
              onSuggestionSelected={[Function]}
              onSuggestionsClearRequested={[Function]}
              onSuggestionsFetchRequested={[Function]}
              renderSuggestion={[Function]}
              renderSuggestionsContainer={[Function]}
              shouldRenderSuggestions={[Function]}
              theme={
                Object {
                  "container": "react-autosuggest__container",
                  "containerOpen": "react-autosuggest__container--open",
                  "input": "react-autosuggest__input",
                  "inputFocused": "react-autosuggest__input--focused",
                  "inputOpen": "react-autosuggest__input--open",
                  "sectionContainer": "react-autosuggest__section-container",
                  "sectionContainerFirst": "react-autosuggest__section-container--first",
                  "sectionTitle": "react-autosuggest__section-title",
                  "suggestion": "react-autosuggest__suggestion",
                  "suggestionFirst": "react-autosuggest__suggestion--first",
                  "suggestionHighlighted": "react-autosuggest__suggestion--highlighted",
                  "suggestionsContainer": "react-autosuggest__suggestions-container",
                  "suggestionsContainerOpen": "react-autosuggest__suggestions-container--open",
                  "suggestionsList": "react-autosuggest__suggestions-list",
                }
              }
            />,
          ],
          "className": "row",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Address",
              "className": "label",
            },
            "ref": null,
            "rendered": "Address",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "clearfix",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "class",
            "props": Object {
              "alwaysRenderSuggestions": false,
              "focusInputOnSuggestionClick": true,
              "getSuggestionValue": [Function],
              "highlightFirstSuggestion": true,
              "id": "1",
              "inputProps": Object {
                "onChange": [Function],
                "placeholder": "Enter Address",
                "value": "street 12",
              },
              "multiSection": false,
              "onSuggestionSelected": [Function],
              "onSuggestionsClearRequested": [Function],
              "onSuggestionsFetchRequested": [Function],
              "renderSuggestion": [Function],
              "renderSuggestionsContainer": [Function],
              "shouldRenderSuggestions": [Function],
              "suggestions": undefined,
              "theme": Object {
                "container": "react-autosuggest__container",
                "containerOpen": "react-autosuggest__container--open",
                "input": "react-autosuggest__input",
                "inputFocused": "react-autosuggest__input--focused",
                "inputOpen": "react-autosuggest__input--open",
                "sectionContainer": "react-autosuggest__section-container",
                "sectionContainerFirst": "react-autosuggest__section-container--first",
                "sectionTitle": "react-autosuggest__section-title",
                "suggestion": "react-autosuggest__suggestion",
                "suggestionFirst": "react-autosuggest__suggestion--first",
                "suggestionHighlighted": "react-autosuggest__suggestion--highlighted",
                "suggestionsContainer": "react-autosuggest__suggestions-container",
                "suggestionsContainerOpen": "react-autosuggest__suggestions-container--open",
                "suggestionsList": "react-autosuggest__suggestions-list",
              },
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "clearfix",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="label"
            >
              Latitude
            </div>,
            <div
              className="clearfix"
            />,
            <input
              className="inputs"
              disabled={true}
              type="text"
              value="51.165691"
            />,
          ],
          "className": "row",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Latitude",
              "className": "label",
            },
            "ref": null,
            "rendered": "Latitude",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "clearfix",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "inputs",
              "disabled": true,
              "type": "text",
              "value": "51.165691",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "clearfix",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="label"
            >
              Longitude
            </div>,
            <div
              className="clearfix"
            />,
            <input
              className="inputs"
              disabled={true}
              type="text"
              value="10.451526000000058"
            />,
          ],
          "className": "row",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Longitude",
              "className": "label",
            },
            "ref": null,
            "rendered": "Longitude",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "clearfix",
            },
            "ref": null,
            "rendered": null,
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "inputs",
              "disabled": true,
              "type": "text",
              "value": "10.451526000000058",
            },
            "ref": null,
            "rendered": null,
            "type": "input",
          },
        ],
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "className": "clearfix",
        },
        "ref": null,
        "rendered": null,
        "type": "div",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <div
              className="delete-btn"
              onClick={[Function]}
            >
              Delete
            </div>,
            <button
              className="save-btn"
              onClick={[Function]}
            >
              Update
            </button>,
          ],
          "className": "row",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Delete",
              "className": "delete-btn",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Delete",
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "Update",
              "className": "save-btn",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": "Update",
            "type": "button",
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <div
            className="row"
          >
            <div
              className="display light-hide"
            >
              No Message
            </div>
          </div>,
          <div
            className="clearfix"
          />,
          <div
            className="row"
          >
            <div
              className="label"
            >
              Title
            </div>
            <div
              className="clearfix"
            />
            <input
              className="inputs"
              onChange={[Function]}
              type="text"
              value="test 1"
            />
          </div>,
          <div
            className="clearfix"
          />,
          <div
            className="row"
          >
            <div
              className="label"
            >
              Address
            </div>
            <div
              className="clearfix"
            />
            <Autosuggest
              alwaysRenderSuggestions={false}
              focusInputOnSuggestionClick={true}
              getSuggestionValue={[Function]}
              highlightFirstSuggestion={true}
              id="1"
              inputProps={
                Object {
                  "onChange": [Function],
                  "placeholder": "Enter Address",
                  "value": "street 12",
                }
              }
              multiSection={false}
              onSuggestionSelected={[Function]}
              onSuggestionsClearRequested={[Function]}
              onSuggestionsFetchRequested={[Function]}
              renderSuggestion={[Function]}
              renderSuggestionsContainer={[Function]}
              shouldRenderSuggestions={[Function]}
              theme={
                Object {
                  "container": "react-autosuggest__container",
                  "containerOpen": "react-autosuggest__container--open",
                  "input": "react-autosuggest__input",
                  "inputFocused": "react-autosuggest__input--focused",
                  "inputOpen": "react-autosuggest__input--open",
                  "sectionContainer": "react-autosuggest__section-container",
                  "sectionContainerFirst": "react-autosuggest__section-container--first",
                  "sectionTitle": "react-autosuggest__section-title",
                  "suggestion": "react-autosuggest__suggestion",
                  "suggestionFirst": "react-autosuggest__suggestion--first",
                  "suggestionHighlighted": "react-autosuggest__suggestion--highlighted",
                  "suggestionsContainer": "react-autosuggest__suggestions-container",
                  "suggestionsContainerOpen": "react-autosuggest__suggestions-container--open",
                  "suggestionsList": "react-autosuggest__suggestions-list",
                }
              }
            />
          </div>,
          <div
            className="clearfix"
          />,
          <div
            className="row"
          >
            <div
              className="label"
            >
              Latitude
            </div>
            <div
              className="clearfix"
            />
            <input
              className="inputs"
              disabled={true}
              type="text"
              value="51.165691"
            />
          </div>,
          <div
            className="clearfix"
          />,
          <div
            className="row"
          >
            <div
              className="label"
            >
              Longitude
            </div>
            <div
              className="clearfix"
            />
            <input
              className="inputs"
              disabled={true}
              type="text"
              value="10.451526000000058"
            />
          </div>,
          <div
            className="clearfix"
          />,
          <div
            className="row"
          >
            <div
              className="delete-btn"
              onClick={[Function]}
            >
              Delete
            </div>
            <button
              className="save-btn"
              onClick={[Function]}
            >
              Update
            </button>
          </div>,
        ],
        "className": "detail-form",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <div
              className="display light-hide"
            >
              No Message
            </div>,
            "className": "row",
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": "No Message",
              "className": "display light-hide",
            },
            "ref": null,
            "rendered": "No Message",
            "type": "div",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "clearfix",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="label"
              >
                Title
              </div>,
              <div
                className="clearfix"
              />,
              <input
                className="inputs"
                onChange={[Function]}
                type="text"
                value="test 1"
              />,
            ],
            "className": "row",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Title",
                "className": "label",
              },
              "ref": null,
              "rendered": "Title",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "clearfix",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "inputs",
                "onChange": [Function],
                "type": "text",
                "value": "test 1",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "clearfix",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="label"
              >
                Address
              </div>,
              <div
                className="clearfix"
              />,
              <Autosuggest
                alwaysRenderSuggestions={false}
                focusInputOnSuggestionClick={true}
                getSuggestionValue={[Function]}
                highlightFirstSuggestion={true}
                id="1"
                inputProps={
                  Object {
                    "onChange": [Function],
                    "placeholder": "Enter Address",
                    "value": "street 12",
                  }
                }
                multiSection={false}
                onSuggestionSelected={[Function]}
                onSuggestionsClearRequested={[Function]}
                onSuggestionsFetchRequested={[Function]}
                renderSuggestion={[Function]}
                renderSuggestionsContainer={[Function]}
                shouldRenderSuggestions={[Function]}
                theme={
                  Object {
                    "container": "react-autosuggest__container",
                    "containerOpen": "react-autosuggest__container--open",
                    "input": "react-autosuggest__input",
                    "inputFocused": "react-autosuggest__input--focused",
                    "inputOpen": "react-autosuggest__input--open",
                    "sectionContainer": "react-autosuggest__section-container",
                    "sectionContainerFirst": "react-autosuggest__section-container--first",
                    "sectionTitle": "react-autosuggest__section-title",
                    "suggestion": "react-autosuggest__suggestion",
                    "suggestionFirst": "react-autosuggest__suggestion--first",
                    "suggestionHighlighted": "react-autosuggest__suggestion--highlighted",
                    "suggestionsContainer": "react-autosuggest__suggestions-container",
                    "suggestionsContainerOpen": "react-autosuggest__suggestions-container--open",
                    "suggestionsList": "react-autosuggest__suggestions-list",
                  }
                }
              />,
            ],
            "className": "row",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Address",
                "className": "label",
              },
              "ref": null,
              "rendered": "Address",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "clearfix",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "class",
              "props": Object {
                "alwaysRenderSuggestions": false,
                "focusInputOnSuggestionClick": true,
                "getSuggestionValue": [Function],
                "highlightFirstSuggestion": true,
                "id": "1",
                "inputProps": Object {
                  "onChange": [Function],
                  "placeholder": "Enter Address",
                  "value": "street 12",
                },
                "multiSection": false,
                "onSuggestionSelected": [Function],
                "onSuggestionsClearRequested": [Function],
                "onSuggestionsFetchRequested": [Function],
                "renderSuggestion": [Function],
                "renderSuggestionsContainer": [Function],
                "shouldRenderSuggestions": [Function],
                "suggestions": undefined,
                "theme": Object {
                  "container": "react-autosuggest__container",
                  "containerOpen": "react-autosuggest__container--open",
                  "input": "react-autosuggest__input",
                  "inputFocused": "react-autosuggest__input--focused",
                  "inputOpen": "react-autosuggest__input--open",
                  "sectionContainer": "react-autosuggest__section-container",
                  "sectionContainerFirst": "react-autosuggest__section-container--first",
                  "sectionTitle": "react-autosuggest__section-title",
                  "suggestion": "react-autosuggest__suggestion",
                  "suggestionFirst": "react-autosuggest__suggestion--first",
                  "suggestionHighlighted": "react-autosuggest__suggestion--highlighted",
                  "suggestionsContainer": "react-autosuggest__suggestions-container",
                  "suggestionsContainerOpen": "react-autosuggest__suggestions-container--open",
                  "suggestionsList": "react-autosuggest__suggestions-list",
                },
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "clearfix",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="label"
              >
                Latitude
              </div>,
              <div
                className="clearfix"
              />,
              <input
                className="inputs"
                disabled={true}
                type="text"
                value="51.165691"
              />,
            ],
            "className": "row",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Latitude",
                "className": "label",
              },
              "ref": null,
              "rendered": "Latitude",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "clearfix",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "inputs",
                "disabled": true,
                "type": "text",
                "value": "51.165691",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "clearfix",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="label"
              >
                Longitude
              </div>,
              <div
                className="clearfix"
              />,
              <input
                className="inputs"
                disabled={true}
                type="text"
                value="10.451526000000058"
              />,
            ],
            "className": "row",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Longitude",
                "className": "label",
              },
              "ref": null,
              "rendered": "Longitude",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "clearfix",
              },
              "ref": null,
              "rendered": null,
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "inputs",
                "disabled": true,
                "type": "text",
                "value": "10.451526000000058",
              },
              "ref": null,
              "rendered": null,
              "type": "input",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "className": "clearfix",
          },
          "ref": null,
          "rendered": null,
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              <div
                className="delete-btn"
                onClick={[Function]}
              >
                Delete
              </div>,
              <button
                className="save-btn"
                onClick={[Function]}
              >
                Update
              </button>,
            ],
            "className": "row",
          },
          "ref": null,
          "rendered": Array [
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Delete",
                "className": "delete-btn",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Delete",
              "type": "div",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Update",
                "className": "save-btn",
                "onClick": [Function],
              },
              "ref": null,
              "rendered": "Update",
              "type": "button",
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getDerivedStateFromProps": true,
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    "disableLifecycleMethods": true,
  },
}
`;
